internal/core/properties/change_tracker.rs

```diff
@@ -91,19 +94,29 @@ impl ChangeTracker {
             let pinned_holder = Pin::new_unchecked(&*_self);
             let _self = _self as *mut BindingHolder<ChangeTrackerInner<T, EF, NF, Data>>;
             let inner = core::ptr::addr_of_mut!((*_self).binding).as_mut().unwrap();
+            assert!(!inner.evaluating);
+            inner.evaluating = true;
             let new_value =
                 super::CURRENT_BINDING.set(Some(pinned_holder), || (inner.eval_fn)(&inner.data));
             if new_value != inner.value {
                 inner.value = new_value;
                 (inner.notify_fn)(&inner.data, &inner.value);
             }
+            if !core::mem::replace(&mut inner.evaluating, false) {
+                core::mem::drop(Box::from_raw(_self));
+            }
```

Comment:
Do I understand correctly that this statement exists to handle the case where drop is called during the evaluation, which will not actually `mem::drop` the box anymore but will delay that until exactly here?

Reply:
Yes, this is correct.  Do you think it needs a comment?

Reply:
Excellent, yes, that helps a bit :)